hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -616,31 +613,41 @@ onOpenNetworkAckSn Environment{party} openState otherParty snapshotSignature sn
 -- __Transition__: 'OpenState' → 'OpenState'
 --
 -- When node receives 'ReqDec' network message it should:
--- - Check there is no decommit in flight
--- - Alter it's state to record what is to be decommitted
--- - Issue a server output 'DecommitRequested' with the relevant utxo
--- - Issue a 'ReqSn' since all parties need to agree in order for decommit be
--- valid to be taken out of a Head.
+-- - Check there is no decommit in flight:
+--   - Alter it's state to record what is to be decommitted
+--   - Issue a server output 'DecommitRequested' with the relevant utxo
+--   - Issue a 'ReqSn' since all parties need to agree in order for decommit to
+--   be taken out of a Head.
+-- - Check if we are the leader
 --
 -- We don't check here if decommit tx can be applied to the confirmed ledger
--- state since this should be done when the snapshot is to be acknowledged. This
--- way we are checking _later_ than we could but it will allow us to not discard
--- decommit UTxOs which are maybe part of the snapshot _in flight_ but we just
--- didn't see the `AckSn` for it yet.
+-- state since this is already done if our party is the decommit _initiator_
+-- (see usage of 'requireValidDecommitTx' when receiving 'Decommit' client
+-- input) and should be done when the snapshot is to be acknowledged.
+--
+-- This way we are checking _later_ than we could but it will allow us to not
+-- discard decommit tx which is maybe part of the snapshot _in flight_ but we
+-- just didn't see the `AckSn` for it yet.
 onOpenNetworkReqDec ::
   IsTx tx =>
+  Environment ->
   OpenState tx ->
   tx ->
   Outcome tx
-onOpenNetworkReqDec openState decommitTx =
-  requireNoDecommitInFlight openState decommitTx $
-    let decommitUTxO = utxoFromTx decommitTx
-     in StateChanged (DecommitRecorded decommitTx)
-          <> Effects
-            [ ClientEffect $ ServerOutput.DecommitRequested headId decommitUTxO
-            , NetworkEffect (ReqSn nextSn (txId <$> localTxs) (Just decommitTx))
-            ]
+onOpenNetworkReqDec env openState decommitTx =
```

Comment:
Must: not override the currently pending `decommitTx` when one is already in flight.

This is what happens if we have a `StateChanged (DecommitRecorded decommitTx)`. We must avoid that by adding a `RequireFailed` here if thre is already a decommit in flight.